src/main/java/com/minecolonies/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -112,8 +112,8 @@ protected AbstractEntityAIStructure(@NotNull final J job)
     private Supplier<AIState> generateSchematicIterator(Function<Structure.SchematicBlock, Boolean> evaluationFunction, AIState nextState)
     {
         // do not replace with method reference, this one stays the same on changing reference for currentStructure
-        Supplier<Structure.SchematicBlock> getCurrentBlock = () -> currentStructure.getCurrentBlock();
-        Supplier<Structure.Result>         advanceBlock    = () -> currentStructure.advanceBlock();
+        Supplier<Structure.SchematicBlock> getCurrentBlock = currentStructure::getCurrentBlock;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'getCurrentBlock' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)